adapter.go

```diff
@@ -690,31 +692,22 @@ func (a *Adapter) AddPolicies(sec string, ptype string, rules [][]string) error
 
 // Transaction perform a set of operations within a transaction
 func (a *Adapter) Transaction(e casbin.IEnforcer, fc func(casbin.IEnforcer) error, opts ...*sql.TxOptions) error {
-	// ensure the transactionMu is initialized
-	if a.transactionMu == nil {
```

Comment:
It would be better to keep the nil check and return error while transactionMu is nil.